### PR TITLE
feat: Autogeneration api documentation files for mkdocs

### LINKED ISSUE #309: template: use auto generation api documentation files for mkdocs
we need to move from a static api/references.md file to something auto generated.

this PR implemented the same for another project and can be used as reference: https://github.com/arxlang/astx/pull/60/files

### PR DESCRIPTION
**Pull request by YurelyCamacho**
## Pull Request description

Auto generation api documentation files for mkdocs

Fixes #309

## How to test these changes

- `...`

## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [X] new feature
- [ ] maintenance

About this PR:

- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more
      complexity.
- [ ] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved
```

### CODE REVIEW

File: src/scicookie/{{cookiecutter.project_slug}}/virtualenvs/conda/dev.yaml [hunk 1]
```diff
@@ -5,6 +5,9 @@ channels:
 dependencies:
   - python >=3.8.1,<4
   - pip
+  - pip:
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/virtualenvs/conda/dev.yaml:8]: this `pip:` section should be at the end of the file and should be included just in the case that the build_system is poetry and the doc engine is mkdocs

**Comment by YurelyCamacho** [comment]
@xmnlab In local `makim tests.unittest` runs correctly and here it doesn't work. I don't understand the problem with the `paginate` installation either.

**Comment by xmnlab** [comment]
oh .. the issue is not inside the tests .. it is in the CI job installation:

https://github.com/osl-incubator/scicookie/actions/runs/10604910289/job/29392594957?pr=311#step:4:83

it happens probably because there is a new release of paginate, and conda install it, but during the installation with poetry, it installs the previous version

pin the previous version (0.5.6) here https://github.com/osl-incubator/scicookie/blob/main/conda/dev.yaml#L15

**Comment by YurelyCamacho** [comment]
@xmnlab I need your help, I have solved several problems that have arisen and now this appears. It is with respect to the build package, in local it shows me: 
`Could not solve for environment specs
The following package could not be installed
└─ build does not exist (perhaps a typo or a missing channel).`

**Comment by YurelyCamacho** [comment]
@xmnlab I did it and the IC does not work

**Comment by YurelyCamacho** [comment]
@xmnlab After several details the CI is running.
I have a doubt about the scripts folder, should it appear only when mkdocs is selected or is it ok like that.

**Comment by xmnlab** [comment]
Great! Yep, for now it is just used by mkdocs .. otherwise it shpuld be removed

**Review by xmnlab — APPROVED** [review]
thanks @YurelyCamacho !

**Pull request merged**